### PR TITLE
feat: Upgrade apisix-base and apisix to openssl3

### PR DESCRIPTION
**Pull request by Revolyssup**

### CODE REVIEW

File: utils/build-common.sh [hunk 1]
```diff
@@ -4,6 +4,19 @@ set -x
 
 ARCH=${ARCH:-`(uname -m | tr '[:upper:]' '[:lower:]')`}
 BUILD_PATH=${BUILD_PATH:-`pwd`}
+OPENSSL3_PREFIX=${OPENSSL3_PREFIX:-`pwd`}
+
+install_openssl_3(){
+    # required for openssl 3.x config
+    cpanm IPC/Cmd.pm
+    git clone https://github.com/openssl/openssl
```
Comment (soulbird) [inline utils/build-common.sh:12]: Should be built based on a fixed version rather than master

File: utils/build-common.sh [hunk 2]
```diff
@@ -39,7 +53,7 @@ build_apisix_base_deb() {
         arch_path="arm64/"
     fi
     DEBIAN_FRONTEND=noninteractive apt-get update
-    DEBIAN_FRONTEND=noninteractive apt-get install -y sudo git libreadline-dev lsb-release libssl-dev perl build-essential
+    DEBIAN_FRONTEND=noninteractive apt-get install -y sudo git libreadline-dev lsb-release libssl-dev perl build-essential
```
Comment (soulbird) [inline utils/build-common.sh:56]: unnecessary modifications

File: dockerfiles/Dockerfile.apisix-base.apk [hunk 3]
```diff
@@ -54,23 +54,6 @@ RUN apk add --no-cache \
 # install latest Rust to build wasmtime
 RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
 
-RUN cd /tmp \
```
Comment (soulbird) [inline dockerfiles/Dockerfile.apisix-base.apk:57]: We no longer maintain the alpine image, you don't need to modify this.
Of course, you can remove it directly in another PR

Comment (Revolyssup) [reply dockerfiles/Dockerfile.apisix-base.apk]: The test that was failing uses this image. I have modified it to install openssl3

**Comment by Revolyssup** [comment]
Build is failing because building apisix expects the shared library with openssl111 [1]

1. https://github.com/apache/apisix/blob/3b16fce060fbac39ee054784213a6bc81df526fc/Makefile#L64

**Comment by Revolyssup** [comment]
CI Passing now. Deleting this PR and creating a new PR with squashed commits

**Pull request closed without merging**